.github/pull_request_template.md

```diff
@@ -0,0 +1,31 @@
+##### SUMMARY
+<!--- Describe the change below, including rationale and design decisions -->
+
+<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
+
+<!--- Please do not forget to include a changelog fragment:
+      https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment
+      No need to include one for docs-only or test-only PR, and for new plugins/modules PRs.
+      Read about more details in CONTRIBUTING.md.
+      -->
+
+##### ISSUE TYPE
+<!--- Pick one below and delete the rest -->
+- Bugfix Pull Request
+- Docs Pull Request
+- Feature Pull Request
+- New Module Pull Request
+- New Plugin Pull Request
+- Test Pull Request
```

Comment:
I like the new types, I think they will make it easier to show what's happening, although this would make more sense if we start gathering stats about it.

A couple of questions though:
* What should be the recommended issue type when doing refactoring? It is not necessarily bug fixing, but it is also not a new feature
* Should we really say "pick one and delete the rest"? I reckon bugfixes and features come (or should come) with tests
* I see you made a distinction between module and plugin for new components. Are you planning on using it for anything, effectively (like something in GHA, or some script to gather stats about PRs or anything of the sort)? I think this might create some confusion with users, given that modules are also under the plugins tree.

Reply:
> What should be the recommended issue type when doing refactoring? It is not necessarily bug fixing, but it is also not a new feature

Hmm, we could also add `Refactoring Pull Request`.

> Should we really say "pick one and delete the rest"? I reckon bugfixes and features come (or should come) with tests

How about "Pick the fitting one(s) and delete the rest"?

Also "Test Pull Request" is more for PRs that **only** do tests (or CI in general), nothing else. There was been no category for them before. Tests as part of features or bugfixes don't need special mention IMO.

> I see you made a distinction between module and plugin for new components.

Mostly because `New Module PR` does not make sense when adding a plugin, and there's nothing else that fits for them. How about having one `New Module/Plugin PR`?

> Are you planning on using it for anything, effectively (like something in GHA, or some script to gather stats about PRs or anything of the sort)? I think this might create some confusion with users, given that modules are also under the plugins tree.

No, it's just that `New Module PR` doesn't fit for plugins.

Reply:
(For stats I would rather look at the modified files in the PR anyway :) )

Reply:
Should be improved in 3cdbb60e0b851ddbd372ad31013670526ec6ce0c.